### PR TITLE
first draft of a CONTRIBUTING document

### PR DESCRIPTION
**Pull request by JaredCorduan**
* Some sections of the main README have been moved to a new CONTRIBUTING
document.
* Added a working conventions section to the CONTRIBUTING document.
* Added performanc section to the CONTRIBUTING document.

I shamelessly pilfered some lovely prose from [Plutus](https://github.com/input-output-hk/plutus/blob/master/CONTRIBUTING.adoc).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+The artifacts in this repository can be built and tested using nix. This is
+additionally used by the Hydra CI to test building, including cross-compilation
+for other systems.
+
+### To add a new Haskell project
+
+To add a new Haskell project, you should do the following:
+
+1. Create the project in the usual way. It should have an appropriate `.cabal` file.
+2. Add the project to the [top-level stack.yaml](./stack.yaml), configuring
```
Comment (nc6) [inline CONTRIBUTING.md]: This is out of date, we don't have a stack.yaml

**Review by nc6 — CHANGES_REQUESTED** [review]
Mostly looks good, but some sections are out of date!

File: CONTRIBUTING.md (hunk 1 above)
Comment (JaredCorduan) [reply CONTRIBUTING.md]: great catch, thanks! it looks like nearly that entire section on adding a new Haskell project was out of date. I made my edit in a separate commit so you could see what small part of it I salvaged.

**Review by nc6 — APPROVED** [review]
Could definitely improve this doc, but this is a nice start!

**Pull request merged**